### PR TITLE
[Web UI] Checks page sorting and silencing

### LINKED ISSUE #1682: [Web UI] Checks Page: Bulk silence
As a user, I should be able to create silences for a selected set of checks.

--- 

screen shot 2018-06-12 at 5 37 57 pm

### LINKED ISSUE #1683: [Web UI] Checks Page: Sort by name
As a user I should be able to sort the checks by their name DESC & ASC.

### LINKED ISSUE #1685: [Web UI] Checks Page: List
As a user, I should be presented with a paginated list of checks associated with the current org/env pair. List items should contain subset of fields relevant to the user, such as the name, command, subscriptions, and interval / schedule (cron).

--- 

screen shot 2018-06-12 at 5 54 28 pm

### PR DESCRIPTION
**Pull request by 10xjs**
## What is this change?

Build out functionality on the checks page. Includes list view styles (minus row details styles), individual and bulk silencing, and sorting.

## Why is this change necessary?

Closes #1685
Closes #1683
Closes #1682

## Does your change need a Changelog entry?

No

## Do you need clarification on anything?

No

## Were there any complications while making this change?

No

## Have you reviewed and updated the documentation for this change? Is new documentation required?

No

### CODE REVIEW

File: dashboard/src/components/partials/ChecksList/ChecksList.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+import TableCell from "@material-ui/core/TableCell";
+import TableRow from "@material-ui/core/TableRow";
+import Typography from "@material-ui/core/Typography";
+
+import { TableListEmptyState } from "/components/TableList";
+import ButtonSet from "/components/ButtonSet";
+
+import Loader from "/components/util/Loader";
+
+import IconButton from "/components/base/IconButton";
```
Comment (jamesdphillips) [inline dashboard/src/components/partials/ChecksList/ChecksList.js]: Did you mean `@material-ui/base/IconButton`? Or did you forgetting to commit a file?

File: dashboard/src/components/partials/ChecksList/ChecksList.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+                        anchorEl={anchorEl}
+                        onClose={close}
+                        options={["NAME"]}
+                        onChangeQuery={onChangeQuery}
+                      />
+                    )}
+                  >
+                    {({ open, ref }) => (
+                      <RootRef rootRef={ref}>
+                        <IconButton onClick={open}>
```
Comment (jamesdphillips) [inline dashboard/src/components/partials/ChecksList/ChecksList.js]: Is an icon button what we really want here?

File: dashboard/src/components/partials/ChecksList/ChecksList.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    limit: undefined,
+    offset: undefined,
+  };
+
+  static fragments = {
+    environment: gql`
+      fragment ChecksList_environment on Environment {
+        checks(
+          limit: $limit
+          offset: $offset
```
Comment (jamesdphillips) [inline dashboard/src/components/partials/ChecksList/ChecksList.js]: Pagination does not appear to the be working. May be related to the `@connection` directive?

File: dashboard/src/components/partials/ChecksList/ChecksListItem.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          />
+        </TableCell>
+        <TableCell style={{ width: "100%" }}>
+          {check.name}
+          <br />
+          {check.command}
+          <br />
+          {check.subscriptions}
+          <br />
+          {check.interval}
```
Comment (jamesdphillips) [inline dashboard/src/components/partials/ChecksList/ChecksListItem.js]: For the sake of anyone using the nightlies lets just use the name in this cell, in it's current state it appears pretty broken / confusing. At least until we build a "details" cell.

File: dashboard/yarn.lock [hunk 5]
```diff
@@ -130,21 +132,19 @@
     jss-props-sort "^6.0.0"
     jss-vendor-prefixer "^7.0.0"
     keycode "^2.1.9"
-    lodash "^4.2.0"
```
Comment (jamesdphillips) [inline dashboard/yarn.lock]: nice

File: dashboard/src/components/partials/ChecksList/ChecksList.js (hunk 3 above)
Comment (10xjs) [reply dashboard/src/components/partials/ChecksList/ChecksList.js]: I was missing a `fetchPolicy` attribute on the `<Query>` tag. Fixed now.

**Pull request merged**